cupyx/scipy/sparse/linalg/_solve.py

```diff
@@ -739,3 +740,243 @@ def _symOrtho(a, b):
+
+    Args:
+        A (ndarray, spmatrix or LinearOperator): The real or complex matrix of
+            the linear system with shape ``(n, n)``.
+        b (cupy.ndarray): Right hand side of the linear system with shape
+            ``(n,)`` or ``(n, 1)``.
+        x0 (cupy.ndarray): Starting guess for the solution.
+        shift (int or float): If shift != 0 then the method solves
+            ``(A - shift*I)x = b``
+        tol (float): Tolerance for convergence.
+        maxiter (int): Maximum number of iterations.
+        M (ndarray, spmatrix or LinearOperator): Preconditioner for ``A``.
+            The preconditioner should approximate the inverse of ``A``.
+            ``M`` must be :class:`cupy.ndarray`,
+            :class:`cupyx.scipy.sparse.spmatrix` or
+            :class:`cupyx.scipy.sparse.linalg.LinearOperator`.
+        callback (function): User-specified function to call after each
+            iteration. It is called as ``callback(xk)``, where ``xk`` is the
+            current solution vector.
+
+    Returns:
+        tuple:
+            It returns ``x`` (cupy.ndarray) and ``info`` (int) where ``x`` is
+            the converged solution and ``info`` provides convergence
+            information.
+
+    .. seealso:: :func:`scipy.sparse.linalg.minres`
+    """
+    A, M, x, b = _make_system(A, M, x0, b)
+
+    matvec = A.matvec
+    psolve = M.matvec
+
+    n = b.shape[0]
+
+    if maxiter is None:
+        maxiter = n * 5
+
+    istop = 0
+    itn = 0
+    Anorm = 0
+    Acond = 0
+    rnorm = 0
+    ynorm = 0
+
+    xtype = x.dtype
+
+    eps = cupy.finfo(xtype).eps
+
+    Ax = matvec(x)
+    r1 = b - Ax
+    y = psolve(r1)
+
+    beta1 = cupy.inner(r1, y)
+
+    if beta1 < 0:
+        raise ValueError('indefinite preconditioner')
+    elif beta1 == 0:
+        return x, 0
+
+    beta1 = cupy.sqrt(beta1)
+    beta1 = beta1.get().item()
+
+    if check:
+        # see if A is symmetric
+        if not _check_symmetric(A, Ax, x, eps):
+            raise ValueError('non-symmetric matrix')
+
+        # see if M is symmetric
+        if not _check_symmetric(M, y, r1, eps):
+            raise ValueError('non-symmetric preconditioner')
+
+    oldb = 0
+    beta = beta1
+    dbar = 0
+    epsln = 0
+    qrnorm = beta1
+    phibar = beta1
+    rhs1 = beta1
+    rhs2 = 0
+    tnorm2 = 0
+    gmax = 0
+    gmin = cupy.finfo(xtype).max
+    cs = -1
+    sn = 0
+    w = cupy.zeros(n, dtype=xtype)
+    w2 = cupy.zeros(n, dtype=xtype)
+    r2 = r1
+
+    while itn < maxiter:
+
+        itn += 1
+        s = 1.0 / beta
+        v = s * y
+
+        y = matvec(v)
+        y -= shift * v
+
+        if itn >= 2:
+            y -= (beta / oldb) * r1
+
+        alpha = cupy.inner(v, y)
+        alpha = alpha.get().item()
+        y -= (alpha / beta) * r2
+        r1 = r2
+        r2 = y
+        y = psolve(r2)
+        oldb = beta
+        beta = cupy.inner(r2, y)
+        beta = cupy.sqrt(beta)
+        beta = beta.get().item()
```

Comment:
`beta` is a scalar so faster on CPU (maybe a couple of percent gain)
```suggestion
        beta = beta.get().item()
        beta = numpy.sqrt(beta)
```